libs-scala/resources/src/main/scala/com/digitalasset/resources/ProgramResource.scala

```diff
@@ -59,5 +64,23 @@ class ProgramResource[T](
 }
 
 object ProgramResource {
-  abstract class SuppressedException extends RuntimeException
+
+  abstract class StartupException(message: String, cause: Throwable)
+      extends RuntimeException
+      with NoStackTrace {
+    def this() = this(null, null)
+
+    def this(message: String) = this(message, null)
+
+    def this(cause: Throwable) = this(null, cause)
+  }
+
+  abstract class SuppressedStartupException(message: String, cause: Throwable)
```

Comment:
What does "suppressed" stand for here?

Reply:
The same suggestion I gave above applies here, but without `message` and `cause`.

Reply:
"Suppressed" stands for "not shown to the user".